janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/StandardJanusGraphTx.java

```diff
@@ -219,12 +219,12 @@ public void close() {
 
         int concurrencyLevel;
         if (config.isSingleThreaded()) {
-            addedRelations = new SimpleBufferAddedRelations();
```

Comment:
`SimpleBufferAddedRelations` and `SimpleAddedRelations` have different implementation of `remove` operation.
`SimpleBufferAddedRelations` - removes all elements with the same id during the remove operation.
`SimpleAddedRelations` - removes only the first one with the same id during the remove operation.